internal/cache/metadata/stat_cache.go

```diff
@@ -51,19 +51,39 @@ type StatCache interface {
 	LookUp(name string, now time.Time) (hit bool, o *gcs.Object)
 }
 
-// Create a new stat cache that holds the given number of entries, which must
-// be positive.
-func NewStatCache(capacity int) (sc StatCache) {
-	sizeOfEntry := entry{}.Size()
-	sc = &statCache{
-		c: lru.NewCache(uint64(capacity) * sizeOfEntry),
+// Create a new bucket-view to the passed shared-cache object.
+// This should be used for dynamic-mount, i.e.
+// mount for a multiple buckets.
+func NewStatCacheBucketView(sc *lru.Cache, bn string) StatCache {
+	return &statCacheBucketView{
+		sharedCache: sc,
+		bucketName:  bn,
 	}
+}
 
-	return
+// Create a new dedicated shared-cache object.
+// This keeps bucketName as "".
+// This should be used for static-mount, i.e.
+// mount for a single bucket.
+func NewStatCache(lruCache *lru.Cache) StatCache {
```

Comment:
can we only keep NewStatCacheBucketView and pass empty string to bucketName ?

Reply:
Removed NewStatCache .